### DISCUSSION TITLE
Create a model plugin that estimates water used by an application

### DISCUSSION THREAD

**Discussion by jmcook1186** [Green-Software-Foundation/hack#28] [Develop an Idea]
## Story
As a user I want to know the environmental impacts of my software beyond carbon. Water is another important environmental concern

## Rationale
IF aims to estimate the environmental impacts of software across many dimensions, but the current set of models only allows us to calculate carbon. It should be possible to calculate the water consumed by an application and output it alongside carbon.

## Scope
This should be a *model plugin* that conforms to the IF [model plugin interface](https://if.greensoftware.foundation/tutorials/how-to-build-models).

## Implementation guidelines
- This model MUST conform to the IF model plugin interface
- The model MUST return a value, water` that measures the amount of water used by an application over a given `duration`
- The model must return `water` in units of...?
- The model must accept `duration` in units of seconds an input.
- The model SHOULD be written in Typescript
- The model MUST have unit tests that demonstrate correct execution
- The model MUST have documentation in the form of a README that explains how to use.

## Difficulty
2/5
This estimate assumes there is a citeable relationship between available inputs and water consumption, in which case model building is expected to be relatively straightforward.

## Impact
3/5 
This will help expand the IF beyond carbon

## Examples and resources
[IF docs](https://if.greensoftware.foundation)
[Model building tutorial](https://if.greensoftware.foundation/tutorials/how-to-build-models)

**Comment by abel-cap** [comment]
@jmcook1186 Great Idea! 
Does it make sense to create a second idea based on biodiversity impact? This is a fairly theoretical approach and I would assume hard to grasp but maybe there is some research regarding negative impact of ICT on biodiversity which could help defining a model.

**Reply by jmcook1186** [reply]
Yes it makes sense to try a biodiversity plugin if you can find some research out there to support some calculations. e.g. can you find input data that you can feed into IF sufficient to drive a calculation with biodiversity loss as an output?
